backend/main/server/resources/Video.py

```
@@ -2,11 +2,28 @@
 from flask import request
 from main.server import db, cache, app
 from main.server.models import Video, VideoSchema
-from flask_jwt import jwt_required
 
 video_schema = VideoSchema()
 videos_schema = VideoSchema(many=True)
 
+def insertVideo(col, data):
+    videoLink = artistLink = username = title = None
+    for i in range(0, len(data)):
+        if col[i] == "videoLink" and data[i]: videoLink = data[i]
+        elif col[i] == "artistLink": artistLink = data[i]
+        elif col[i] == "username": username = data[i]
+        elif col[i] == "title": title = data[i]
+        else: continue
```

We can afford the extra lines, so separate the operations under the conditionals to be more pythonic.

I would prefer if we can fail faster with this code. 2 suggestions:
1. Instead of iterating through, transform the input to be a `dict` with keys and values. Or if you prefer, transform it after you pass it here.
2. Make the marshmallow schema do the heavy lifting of validating your data. Right now with this format, we no longer confirm the data we try to load into the db.

---

OK, I'll have to look at how you did the dict on the original csv uploader script.

---

can you take a look at the latest commit and see if it's satisfactory? I didn't use dicts, but I made it not so ugly as this.